src/pages/NewChatPage.js

```diff
@@ -108,17 +108,21 @@ class NewChatPage extends Component {
             }
         }
 
+        const filterText = _.reduce(this.state.selectedOptions, (str, {login}) => `${str} ${login}`, '');
```

Comment:
This just takes all the filtered users and puts their name in a big string right? How come we do this over just putting them in an array and doing .includes?

Reply:
This O(n). That would be O(n*n)

Reply:
How come? I wasn't able to find details about runtime complexity about string.includes() but I assume string.includes() in the worst case still searches the entire string

I found this [SO](https://stackoverflow.com/questions/48760542/time-complexity-of-the-includes-method-in-javascript) on array.includes() which says it uses linear search and so the worst scenario it is O(n).

So if that is the case the runtime wouldn't be different

Reply:
Ok. I will do this. It's no biggy. It can never hurt the app performance in any way.

Reply:
but this is the breakout: 
=> 111 O(n)
=> 112 O(n)

= O(n). 

New changes. 
```js
_.filter(this.state.recentReports, ({login}) => !this.state.selectionOptions.find({slogin} => login === slogin));
```
There are two nested loops thus O(n*n).

Reply:
Do you still think change is needed? Let me know.

Reply:
Thanks for exploring this with me. I think they're both O(n*n) since _.reduce is also O(n) along with the string.includes O(n).

Though my biggest concern about it was that using a space delimiter is that it might lead to some edge cases with spaces in logins versus something more explicit like an array.  But after thinking about it some more this might not be that big of a deal since the logins are emails

Reply:
That being said, thanks again for looking into this but I think what you have here will be okay for our use case

Reply:
Yeah. Overlooked it. Thanks for understanding. I will update the screenshot in few mintues.